### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by Turbo87**
This PR adjusts the assertion code to generate better error messages. See the unit test changes for examples.

### CODE REVIEW

**Comment by Turbo87** [comment]
@trabus any comments?

**Comment by trabus** [comment]
LGTM

I originally just copied over what was in ember-cli without trying to change anything. Might want to get comments from @rwjblue and @stefanpenner though.

File: test/assert-file-to-not-exist-test.js [hunk 1]
```diff
@@ -7,8 +8,13 @@ describe('assert-file-to-not-exist', function() {
   });
 
   it('should throw if file exists', function() {
-    expect(function() {
+    var err = catchError(function() {
```
Comment (stefanpenner) [inline test/assert-file-to-not-exist-test.js]: im curious how this improves upon expect's existing `throw` helper

Comment (Turbo87) [inline test/assert-file-to-not-exist-test.js]: with chai you can only assert the message and type of the exception, but not other properties like actual and expected.

Comment (stefanpenner) [inline test/assert-file-to-not-exist-test.js]: can we make this an extension to chai instead? `expect(x).to.catchError(....)`

Comment (Turbo87) [inline test/assert-file-to-not-exist-test.js]: I guess we could override the throw property to handle a function argument. I'll have a look.

Comment (stefanpenner) [inline test/assert-file-to-not-exist-test.js]: lets not replace an existing helper, but rather lets add a new one.

Comment (Turbo87) [inline test/assert-file-to-not-exist-test.js]: iirc we can just add that behavior instead of overwriting the original

Comment (stefanpenner) [inline test/assert-file-to-not-exist-test.js]: :+1:

Comment (Turbo87) [inline test/assert-file-to-not-exist-test.js]: done, seems to work fine. new API looks like this:

``` js
    expect(function() {
      assertFile('test/fixtures/missing.txt');
    }).to.throw(function(err) {
      expect(err.toString()).to.equal('AssertionError: expected \'test/fixtures/missing.txt\' to exist');
      expect(err.actual).to.not.exist;
      expect(err.expected).to.not.exist;
    });
```

**Pull request merged**